xpbooster.java

```diff
@@ -50,145 +57,66 @@ public void onStatChanged(StatChanged statChanged) {
 
 		if (skill == Skill.OVERALL) return;
 
-		int realpreviousxp = realXpTracker.get(skill);
-		int fakepreviousxp = fakeXpTracker.get(skill);
+		int realPreviousXp = realXpTracker.get(skill);
+		int fakePreviousXp = fakeXpTracker.get(skill);
 
-		int realcurrentxp = currentXp;
-		int realXpGained = realcurrentxp - realpreviousxp;
-		int fakecurrentxp;
+		int realCurrentXp = currentXp;
+		int realXpGained = realCurrentXp - realPreviousXp;
+		int fakeCurrentXp;
 
 		//For first startup when stats are changed from 0 to your current stat, couldn't be bothered to figure out a
 		//smarter (read less lazy) way of detecting this.
 		if (realXpGained > 1000){
-			fakecurrentxp = currentXp;
+			fakeCurrentXp = currentXp;
 		} else {
-			//normal operation
-			fakecurrentxp = fakepreviousxp + (realXpGained * 10000);
+			fakeCurrentXp = fakePreviousXp + (realXpGained * 10000);
 		}
 
-		realXpTracker.put(skill, realcurrentxp);
-		fakeXpTracker.put(skill, fakecurrentxp);
+		realXpTracker.put(skill, realCurrentXp);
+		fakeXpTracker.put(skill, fakeCurrentXp);
 	}
 
 	@Subscribe
 	public void onBeforeRender(BeforeRender tick)
 	{
-		System.out.println("real attack level: " + Experience.getLevelForXp(realXpTracker.get(Skill.ATTACK)));
-		System.out.println("fake attack level: " + Experience.getLevelForXp(fakeXpTracker.get(Skill.ATTACK)));
-		//Hope you enjoy this indentation
-		for (int i = 7995410; i < 7995417; i++){
-			Widget xp = client.getWidget(i);
-			if (xp != null) {
-				if (xp.getChildren() != null) {
-					for (Widget child : xp.getChildren()){
-						if (child != null) {
-							if (child.getText() != null) {
-								if (!child.getText().isEmpty())
-									if (!child.getText().endsWith("0000")) {
-										child.setText(child.getText() + "0000");
-									 	xp.setOriginalWidth(xp.getOriginalWidth() + 30);
-										//Surely one of these will revalidate it right?
-										xp.revalidate();
-										child.revalidate();
-										xp.getChildren()[1].revalidate();
-									}
-								}
-							}
-						}
-					}
-				}
-			}
-
-		for (Skill skill : Skill.values()) {
-			if (skill != Skill.OVERALL) {
-				updateSkillLevel(skill);
+		final int firstWidgetId = 7995410;
+		final int lastWidgetId = 7995417;
+
+		for (int widgetId = firstWidget; widgetId < lastWidgetId; widgetId++){
+			Widget xpWidget = client.getWidget(widgetId);
+
+			if (xpWidget == null || xpWidget.getChildren() == null) continue;
```

Comment:
[Guard clauses](https://deviq.com/design-patterns/guard-clause)! Pyramid of doom no more!